football_explorer/explorer.py

```diff
@@ -9,27 +8,21 @@ def __init__(self, csv_file_name):
     def all(self):
         with open(self.csv_file_name) as f:
             footballers = csv.reader(f)
-            all_footballers = []
             for footballer in footballers:
-                all_footballers.append(Player(*footballer))
-        return all_footballers
+                yield Player(*footballer)
 
     def search(self, country=None, year=None, age=None, position=None):
         search_args = locals()
-        player_idx = {
-          'country': 6,
-          'year':8,
-          'age':3,
-          'position':1
+        player_attr = {
+            'country': 'country',
+            'year': 'year',
+            'age': 'date_of_birth',
+            'position': 'position'
         }
-        filter_dict = {player_idx[key]: search_args[key] for key in player_idx if search_args[key] is not None}
+        filter_dict = {player_attr[key]: search_args[key] for key in player_attr if search_args[key] is not None}
```

Comment:
I like the use of dictionary comprehension here it really clever. The only thing it needs are some comments as it took me some time to figure out what was going on here.